robottelo/performance/graph.py

```diff
@@ -0,0 +1,121 @@
+"""Test utilities for generating charts for both Candlepin and Pulp tests"""
+import pygal
+
+
+def generate_bar_chart_stat(stat_dict, head, filename, lengend):
+    """Generate Bar chart for stat on concurrent subscription
+
+    :param dict stat_dict: The dictionary containing min/median/max/std
+    :param str head: Titile of charts
+    :param str filename: The name of output svg chart
+    :param str lengend: The lengend of generated bar charts
+
+    """
+    bar_chart = pygal.Bar()
+    bar_chart.title = head
+    bar_chart.x_labels = ('min', 'mean', 'max', 'std')
+    bar_chart.x_title = 'Statistics'
+    bar_chart.y_title = 'Time (s)'
+    for key in range(len(stat_dict)):
+        bar_chart.add('{0}-{1}'.format(lengend, key), stat_dict.get(key))
+    bar_chart.render_to_file(filename)
+
+
+def generate_line_chart_bucketized_stat(
+        stat_dict,
+        head,
+        filename,
+        bucket_size,
+        num_buckets):
+    """Generate Line chart for stat on concurrent subscription
+
+    Take a dictionary of statistic values on each client. For example::
+
+        0: [min, | median,max,std]
+        1: [min, | median,max,std]
+        2: [min, | median,max,std]
+        ...
+        9: [min, | median,max,std],
+
+        it would slice out all min, then all median, all max, and all std
+        sequentially, form 4 lines on the chart, where each line represented
+        by 10 timing points.
+
+    :param dict stat_dict: The dictionary containing min/median/max/std
+    :param str head: Titile of charts
+    :param str filename: The name of output svg chart
+    :param int bucket_size: Number of timing values grouped in each bucket
+    :param int num_buckets: Number of buckets of the test; default 10
+
+    """
+    if bucket_size == 0:
+        return
+
+    # parameters for formats on charts
+    lengend = ('min', 'median', 'max', 'std')
+    buckets = [
+        '{0}-{1}'.format(bucket_size * i + 1, bucket_size * (i + 1))
+        for i in range(num_buckets)
+    ]
+    line_chart = pygal.Line()
+    line_chart.title = head
+    line_chart.x_labels = buckets
+    line_chart.x_title = 'Buckets'
+    line_chart.y_title = 'Time (s)'
+    for count, stat in enumerate(lengend):
+        # count min/median/max/std separately
+        tmp_list = []
+        # accumulate stat item of each client
+        for key, valuelist in stat_dict.iteritems():
+            tmp_list.append(valuelist[count])
+        # add list to chart
+        line_chart.add(stat, tmp_list)
```

Comment:
The `key` variable is unused, so drop it. `iteritems` is not available in Python 3, so do not use it. Consider using generators instead of lists, as they are more efficient and just as easy to use. Comprehensions are preferable to short-lived lists.

How about this?

```python
    for count, stat in enumerate(lengend):
        line_chart.add((value[count] for value in stat_dict.values()))
```